src/main/java/de/rondiplomatico/nds/floodFill.java

```diff
@@ -0,0 +1,271 @@
+					}
+				} else if (vline && ((ff[row][col+1] != bg) || (ff[row][col-1] != bg)) && (ff[row][col] != bg)) {
+					// end of horizontal line
+					currStopY 		= row;
+					vlineEnded 		= true;
+				} else {
+					vline = false;
+					vlineEnded = false;
+				}
+				if (vline && vlineEnded) {
+					// fill line and reset booleans
+					System.out.println(">>>>>>INFO: Filling vertical line.");
+					for (int idx = currStartY; idx < currStopY; idx++) {
+						ff[idx][col]	= fill;
+					}
+					vline 				= false;
+					vlineEnded 			= false;
+				}
+			}
+		}
+		// second, find horizontal lines
+		boolean hline 					= false;
+		boolean hlineEnded				= false;
+		for (int row = 1; row < dimX-1; row++) {
+			hline 						= false;
+			hlineEnded 					= false;
+			for (int col = 1; col < dimY-1; col++) {
+				// minimum requirement: top and bottom neighbors are not background
+				if ((ff[row+1][col] != bg) && (ff[row-1][col] != bg)) {
+					if ((ff[row][col] == bg)
+						&& (ff[row][col-1] != bg)) {
+						// beginning of horizontal line
+						currStartY 		= col;
+						hline 			= true;
+					} else if (hline && (ff[row][col] == bg)) {
+						// vertical line continues
+					} else if (hline && (ff[row][col] != bg)) {
+						// end of vertical line
+						currStopY 		= col;
+						hlineEnded 		= true;
+					}
+				} else if (hline && ((ff[row+1][col] != bg) || (ff[row-1][col] != bg)) && (ff[row][col] != bg)) {
+					// end of vertical line
+					currStopY 		= col;
+					hlineEnded 		= true;
+				} else {
+					hline = false;
+					hlineEnded = false;
+				}
+				if (hline && hlineEnded) {
+					// fill line and reset booleans
+					System.out.println(">>>>>>INFO: Filling horizontal line.");
+					for (int idx = currStartY; idx < currStopY; idx++) {
+						ff[row][idx]	= fill;
+					}
+					hline 				= false;
+					hlineEnded 			= false;
+				}
+			}
+		}
+		
+		return;
+	}
+	
+	/*
+	 * Set initial point for flood fill algorithm
+	 * 
+	 * @todo need to do more extensive checks if we can always find an initial point if we only check the right half of the box
+	 */
+	public void setStartingPoint() {
+		startAtX 						= (int) (0.5 * dimX);
+		startAtY 						= 0;
+		// we're starting on background, skip that..
+		while (ff[startAtX][startAtY] == bg) {
+			startAtY++;
+		}
+		// we're continuing on an edge, skip that..
+		while (ff[startAtX][startAtY] == bound) {
+			startAtY++;
+		}
+		for (int row = startAtY; row < dimY; row++, startAtY++) {
+			int rowSum 					= 0;
+			for (int col = startAtX; col < dimX; col++) {
+				if (ff[col][row-1] != ff[col][row]) {
+					rowSum 					= rowSum + ff[col][row];
+				}
+			}
+			if ((rowSum % 2) == 1) {
+				// we've found a point that's inside
+				break;
+			}
+		}
+		return;
+	}
+	
+	/*
+	 * Get initial point for flood fill algorithm
+	 * 
+	 */
+	public int[] getStartingPoint() {
+		return new int[] {startAtX, startAtY};
+	}
+
+	/*
+	 * Get flood filled image
+	 * 
+	 */
+	public int[][] getFloodFill(){

```

Comment:
if i run
```java
 floodFill f = new floodFill();
 int[][] ff = f.getFloodFill();
 int a = ff.length; //-> NullPointerException.
```
Always return useful values for public access methods or state what to expect. what needs to be called first before this contains useful values? et

Reply:
The first line will cause a compile-time error because there's not constructor without arguments. So this should never happen? I've added a comment in the constructor that we expect a non-empty image to be supplied.